go/arrow/array/fixedsize_binary_test.go

```diff
@@ -0,0 +1,37 @@
+package array
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/apache/arrow/go/arrow"
+	"github.com/apache/arrow/go/arrow/memory"
+)
+
+func TestFixedSizeBinary(t *testing.T) {
+	mem := memory.NewCheckedAllocator(memory.NewGoAllocator())
+	defer mem.AssertSize(t, 0)
+
+	dtype := arrow.FixedSizeBinaryType{ByteWidth: 7}
+	b := NewFixedSizeBinaryBuilder(mem, dtype)
+
+	values := [][]byte{
+		[]byte("7654321"),
+		nil,
+		[]byte("AZERTYU"),
+	}
+	b.AppendValues(values, []bool{true, false, true})
+
+	a := b.NewFixedSizeBinaryArray()
```

Comment:
ditto: perhaps add a "dummy" pair of `Retain/Release` ?

Reply:
also: perhaps use the `NewArray` API (with a type-cast to `*FixedSizeBinary`) to also make sure `NewArray` works correctly.